ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding_pybind.hpp

```diff
@@ -47,7 +47,7 @@ void bind_tilize_with_val_padding(py::module &module) {
             [](const OperationType &self,
                const ttnn::Tensor &input_tensor,
                const tt::tt_metal::LegacyShape &output_tensor_shape,
-               float value,
+               const PadValue value,
```

Comment:
This feels error prone to me.
In most cases we want to pass the float here, but if user doesn't write number in float format in python, 
it will get the wrong thing. I think this will happen often.
One way to mitigate this would be to add an assert to check that tensor and PadValue are in sync, but still seems like
an API that will be hard to use.

Reply:
ah ya I modified it to cast if there is a mismatch between the get_dtype() and the type in the variant.